tests/modules/fgbio/groupreadsbyumi/nextflow.config

```diff
@@ -2,4 +2,8 @@ process {
 
     publishDir = { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" }
 
+    withName: FGBIO_GROUPREADSBYUMI {
+        ext.strategy = 'Adjacency'
+    }
```

Comment:
Didn't you specify the default strategy to be `Adjacency` by default in the module already?
Instead of having `ext.strategy = 'Adjacency'` can we add `-s Adjacency` to `ext.args` or even to `ext.args2`?

Reply:
I did, but wanted to be explicit in the test.
I assume we are using `ext.args` and `ext.args2` for optional arguments (at least this what I could gather from the other modules). Instead `-s` is a *mandatory* argument in groupreadsbyumi, i.e. always required: this means the `-s` option in the code cannot be left to the user, only its content.
http://fulcrumgenomics.github.io/fgbio/tools/latest/GroupReadsByUmi.html
what do you think?

Reply:
@drpatelh @mahesh-panchal what's your take on that?

Reply:
The idea is usually to check args for any mandatory parameters to make sure they're set, otherwise add them:
https://github.com/nf-core/modules/blob/3b366c7c6aac446c1a4ea7c2016092344633b2ec/modules/seqtk/sample/main.nf#L36-L38

Reply:
happy to change this way. 
but then we might want to change our guidelines https://nf-co.re/developers/adding_modules#general where it is specified that "All *non-mandatory* command-line tool options MUST be provided as a string via the $args variable" :)

Reply:
to be more explicit :)
do we want to (a) revise the guidelines and use `ext.args` also for mandatory arguments using the solution proposed by @mahesh-panchal or (b) we want to handle mandatory arguments with other variables inside `ext.` object (I have no preference on names etc here of course).
I would be in favour of keeping `ext.args` for non mandatory things (I like to be explicit in coding), but since this is relevant for everyone, a solution people are happy with is good for me :)

Reply:
Well there are at least two ways to provide mandatory arguments. I think the way envisioned by most workflows is a workflow parameter `params.tool_option` which is then passed via an `input:`. The other way is to automatically provide a default for perhaps an option that is uncommonly changed by appending it to `$args`. Also it would be helpful to describe in the docs how to handle mandatory args, but the current statement on non-mandatory args is still fine in my opinion. I think avoiding over-use of the `process.ext` should also be encouraged, in favour of one of the two methods above.

Reply:
ok for me then, I'm all for this solution @mahesh-panchal @lescai

Reply:
which solution?

Reply:
not the `ext.args` one, so passed over with an input

Reply:
Sorry about that